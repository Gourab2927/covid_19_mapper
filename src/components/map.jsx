import React, { Component } from "react";
import {
  Map,
  GoogleApiWrapper,
  Marker,
  InfoWindow,
  Circle,
} from "google-maps-react";
import styles from "./assets/mapStyle.json";

export class MapContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      center: { lat: 40.4929, lng: 15.5553 },
      isMarkerShowing: true,
      activeMarker: null,
      selectedPlace: {},
      activeMarker: {},
    };
  }

  // onMapClicked = (mapProps, map, event) => {
  // console.log("mapProps", mapProps);
  // console.log("map", map);
  // console.log("event", event);
  // if (data.marginBounds.nw.lat >= 85 || data.marginBounds.se.lat <= -85) {
  //   this.setState({
  //     center: [this.state.center[0] + 0.00100, this.state.center[1]],
  //   });
  // }
  // this.setState({
  //   isMarkerShowing: false,
  //   activeMarker: ""
  // });
  // };

  onMarkerClicked = (props, marker, event) => {
    this.setState({
      isMarkerShowing: true,
      selectedPlace: props,
      activeMarker: marker,
    });
  };

  generateMarkers = () => {
    return this.props.countries.map((country, i) => {
      return (
        <Marker
          id={country.country}
          key={i}
          position={country.center}
          onClick={this.onMarkerClicked}
          icon={
            country.us
              ? "http://maps.google.com/mapfiles/ms/icons/blue.png"
              : "http://maps.google.com/mapfiles/ms/icons/purple.png"
          }
        ></Marker>
      );
    });
  };

  render() {
    return (
      <div className="Map">
        <Map
          google={this.props.google}
          styles={styles}
          initialCenter={this.state.center}
          zoom={2.2}
          minZoom={2.2}
          maxZoom={12}
          // onClick={this.onMapClicked}
          disableDefaultUI={true}
        // onClick={this.onChange}
        >
          <InfoWindow
            marker={this.state.activeMarker}
            visible={this.state.isMarkerShowing}
          >
            {this.props.countries.map((country) => {
              if (country.country === this.state.selectedPlace.id) {
                if (country.us) {
                  return (
                    <div>
                      <h6>{country.country}</h6>
                      Total Cases: {country.confirmed}
                      <br></br>
                      Total Deaths: {country.deaths}
                    </div>
                  );
                } else {
                  return (
                    <div>
                      <h6>{country.country}</h6>
                      Total Cases: {country.confirmed}
                      <br></br>
                      Total Deaths: {country.deaths}
                      <br></br>
                      Total Recoveries: {country.recovered}
                      <br></br>
                      Active Cases: {country.activeCases}
                      <br></br>
                      Critical Cases: {country.criticalCases}
                      <br></br>
                      New Cases: {country.newCases}
                      <br></br>
                      New Deaths: {country.newDeaths}
                      <br></br>
                      Cases per Million: {country.perOneMillion}
                    </div>
                  );
                }
              }
            })}
          </InfoWindow>
          {this.generateMarkers()}
        </Map>
      </div>
    );
    // };
  }
}

export default GoogleApiWrapper({
  apiKey: process.env.REACT_APP_GOOGLE_API,
})(MapContainer);
