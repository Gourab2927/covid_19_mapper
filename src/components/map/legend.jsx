import React, { Component } from 'react';


class Legend extends Component {
  state = {}
  render() {
    return (
      <div id="card" >
        <div>

          <div id="colour"><svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px"
            width="30" height="30"
            viewBox="0 0 172 172"
            styles=" fill:#000000;"><g fill="none" fill-rule="nonzero" ><path d="M0,172v-172h172v172z" fill="none"></path><g fill="#dc3645"><path d="M86,157.66667c-39.41667,0 -71.66667,-32.25 -71.66667,-71.66667c0,-39.41667 32.25,-71.66667 71.66667,-71.66667c39.41667,0 71.66667,32.25 71.66667,71.66667c0,39.41667 -32.25,71.66667 -71.66667,71.66667z">
            </path></g></g></svg><svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px"
              width="30" height="30"
              viewBox="0 0 172 172"
              styles=" fill:#000000;"><g fill="none" fill-rule="nonzero" ><path d="M0,172v-172h172v172z" fill="none"></path><g fill="#ffc108"><path d="M86,157.66667c-39.41667,0 -71.66667,-32.25 -71.66667,-71.66667c0,-39.41667 32.25,-71.66667 71.66667,-71.66667c39.41667,0 71.66667,32.25 71.66667,71.66667c0,39.41667 -32.25,71.66667 -71.66667,71.66667z">
              </path></g></g></svg>Colour: Red if the countries Case Fatality Rate is greater than the global average</div>
        </div>
        <br></br>
        <br></br>
        <div id="size"><svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px"
          width="30" height="30"
          viewBox="0 0 172 172"
          styles=" fill:#000000;"><g fill="none" fill-rule="nonzero" ><path d="M0,172v-172h172v172z" fill="none"></path><g fill="#dc3645"><path d="M86,157.66667c-39.41667,0 -71.66667,-32.25 -71.66667,-71.66667c0,-39.41667 32.25,-71.66667 71.66667,-71.66667c39.41667,0 71.66667,32.25 71.66667,71.66667c0,39.41667 -32.25,71.66667 -71.66667,71.66667z">
          </path></g></g></svg><svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px"
            width="15" height="15"
            viewBox="0 0 172 172"
            styles=" fill:#000000;"><g fill="none" fill-rule="nonzero" ><path d="M0,172v-172h172v172z" fill="none"></path><g fill="#dc3645"><path d="M86,157.66667c-39.41667,0 -71.66667,-32.25 -71.66667,-71.66667c0,-39.41667 32.25,-71.66667 71.66667,-71.66667c39.41667,0 71.66667,32.25 71.66667,71.66667c0,39.41667 -32.25,71.66667 -71.66667,71.66667z">
            </path></g></g></svg>Size</div>

      </div >
    );
  }
}

export default Legend;