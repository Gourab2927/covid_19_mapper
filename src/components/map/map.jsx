import React, { Component } from "react";
import {
  withGoogleMap,
  GoogleMap,
  InfoWindow,
  Marker,
  Circle,
} from "react-google-maps";
import styles from "./../assets/mapStyle.json";
import MapInfo from "./mapinfo";

class MapContainer extends Component {
  constructor(props) {
    super(props);
    this.map = React.createRef();
    this.circle = React.createRef();
    this.countryElement = React.createRef();
    this.getZoom = this.getZoom.bind(this);
    this.idleCalled = false;

    this.state = {
      center: { lat: 40.4929, lng: 15.5553 },
      isMarkerShowing: true,
      selectedPlace: {},
      activeMarker: {},
    };
  }
  getCircle(magnitude) {
    return {
      path: this.circle.current,
      scale: Math.pow(2, magnitude) / 2,
    };
  }

  getZoom() {
    return this.map.current.getZoom();
  }

  // handleIdle() {
  //   // console.log(this.idleCalled);
  //   if (!this.idleCalled) {
  //     console.log(this.circle);
  //     if (this.circle.current !== null) {
  //       // console.log(this.circle.current.getRadius());
  //       this.changeCircleRadius();
  //       // this.map.current.this.idleCalled = false;
  //     }
  //   }
  // }

  // changeCircleRadius() {
  //   if (this.getZoom() <= 4) {
  //     return 25000;
  //   } else if (this.getZoom() > 3 && this.getZoom() <= 8) {
  //     return 12500;
  //   } else {
  //     return 10000;
  //   }
  // }

  onMarkerClicked = (marker) => {
    this.countryElement.current.changeCountry(marker);
  };

  onCircleClicked = (props) => {
    console.log(props);
  };

  render() {
    const GoogleMapExample = withGoogleMap((props) => (
      <GoogleMap
        ref={this.map}
        defaultCenter={this.state.center}
        defaultZoom={2.25}
        options={{
          disableDefaultUI: true,
          styles: styles,
          minZoom: 2,
          maxZoom: 10,
          zoomControl: true,
        }}
        onIdle={this.handleIdle}
      >
        {this.props.countries.map((country, i) => (
          <Circle
            ref={this.circle}
            defaultCenter={country.center}
            radius={
              (country.confirmed / this.props.total[0]) * 100 <= 1.5 || NaN
                ? 60000
                : 160000
            }
            onClick={() => this.onMarkerClicked(country.country)}
            options={
              country.us
                ? {
                    strokeColor: "",
                    strokeOpacity: 0.8,
                    strokeWeight: 0.2,
                  }
                : 
                (country.deaths / country.confirmed) * 100 <= 2.5
                ? {
                    fillColor: "#28A745",
                    fillOpacity: 0.5,
                    strokeOpacity: 0.8,
                    strokeWeight: 0,
                  }
                : (country.deaths / country.confirmed) * 100 <= 5
                ? {
                    fillColor: "#FFC108",
                    fillOpacity: 0.5,
                    strokeOpacity: 0.8,
                    strokeWeight: 0,
                  }
                : {
                    fillColor: "#DC3645",
                    fillOpacity: 0.5,
                    strokeOpacity: 0.8,
                    strokeWeight: 0,
                  }
            }
          ></Circle>
        ))}
      </GoogleMap>
    ));

    return (
      <div>
        <div>
          <GoogleMapExample
            containerElement={<div id="container" />}
            mapElement={<div id="map" />}
          />
        </div>
        <div id="mapinfo">
          <MapInfo
            ref={this.countryElement}
            countriesArray={this.props.countries}
          />
        </div>
      </div>
    );
  }
}

export default MapContainer;
