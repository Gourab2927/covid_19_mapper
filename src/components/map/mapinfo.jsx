import React, { Component } from 'react';


class MapInfo extends Component {
  constructor(props){
    super(props)
    this.state = { 
      marker: "Iceland"
    }
  }
  

  // componentDidMount(){
    
  //   this.setState({
  //     marker: this.props.country
  //   })
  // }

  // componentDidUpdate(props, prevProps){
  //   if(props.country !== prevProps.country){
  //     this.setState({
  //       marker: props.country
  //     })
  //   }
  // }

  filterCountries = () => {
    var a = this.props.countriesArray.filter(country => country.country === this.state.marker)
      console.log(a)
  }

  render() { 
    return ( 
    <div>
      {this.props.countriesArray.map(country => {
        if(country.country === this.state.marker) {
         return `${country.country} Confirmed: ${country.cases} Deaths: ${country.deaths} Recovered: ${country.recovered}`
        }
      })}
    </div> );
  }
}
 
export default MapInfo;