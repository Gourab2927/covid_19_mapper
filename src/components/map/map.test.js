import * as React from 'react';
import {ReactDOM, unmountComponentAtNode} from "react-dom";
import { render } from '@testing-library/react';
import { shallow, simulate } from 'enzyme';
import {MapContainer, Map} from './map'
import Marker from 'google-maps-react'
import renderer from 'react-test-renderer';
import MockedMap from './map'
import { act } from "react-dom/test-utils"



it('does something on marker click', () => {    
  const onMarkerClick = jest.fn();
  const component = shallow(<MapContainer countries={[]} onMarkerClick={onMarkerClick}/>); 
  const marker = component.find(Marker)
  expect(marker.simulate('click').first().length).toEqual(1)
});

