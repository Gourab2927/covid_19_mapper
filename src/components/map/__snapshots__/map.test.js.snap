// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <div
      id="container"
    >
      <div
        id="map"
      />
      <div />
    </div>
  </div>
  <div
    id="mapinfo"
  >
    <div
      className="bg-warning text-white"
      id="card"
    />
  </div>
</div>
`;
